lib/openstack_query/runners/runner_wrapper.py

```diff
@@ -34,80 +34,112 @@ def __init__(self, marker_prop_func: PropFunc, connection_cls=OpenstackConnectio
-                logger.debug("parsing subset of %s items", len(from_subset))
-                start = time.time()
-                resource_objects = self._parse_subset(conn, from_subset)
+                logger.info("'from_subset' meta param given - running query on subset")
+                resource_objects = self._parse_subset(
+                    conn=conn,
+                    subset=from_subset,
+                )
                 logger.debug(
-                    "parsing complete - time elapsed: %s seconds", time.time() - start
+                    "subset parsed. (time elapsed: %0.4f seconds)", time.time() - start
                 )
             else:
                 logger.info("running query using openstacksdk and server-side filters")
-                server_side_filters_log_str = "none (getting all)"
-                kwarg_log_str = "none"
-                if server_side_filters:
-                    server_side_filters_log_str = "\n\t\t".join(
-                        [f"{key}: '{val}'" for key, val in server_side_filters.items()]
-                    )
-                if kwargs:
-                    kwarg_log_str = "\n\t\t".join(
-                        [f"{key}: '{val}'" for key, val in kwargs.items()]
-                    )
-
-                logger.debug(
-                    "calling run_query with parameters "
-                    "\n\tserver_side_filters: "
-                    "\n\t\t%s "
-                    "\n\trun_meta_kwargs: "
-                    "\n\t\t%s",
-                    server_side_filters_log_str,
-                    kwarg_log_str,
-                )
-                start = time.time()
-                query_meta_params = {}
-                if kwargs:
-                    query_meta_params = self._parse_meta_params(conn, **kwargs)
-                resource_objects = self._run_query(
-                    conn, server_side_filters, **query_meta_params
+                resource_objects = self._run_with_openstacksdk(
+                    conn=conn, server_filters=server_side_filters, **kwargs
                 )
-                logger.info(
-                    "server-side query complete - time elapsed: %s seconds",
+                logger.debug(
+                    "server-side quer(y/ies) completed - found %s items. (time elapsed: %0.4f seconds)",
+                    len(resource_objects),
                     time.time() - start,
                 )
-                logger.debug("server-side query found %s items", len(resource_objects))
 
-        if client_side_filter_func and not server_side_filters:
+        if client_side_filters:
             logger.info("applying client side filters")
-            resource_objects = self._apply_client_side_filter(
-                resource_objects, client_side_filter_func
+            resource_objects = self._apply_client_side_filters(
+                items=resource_objects, filters=client_side_filters
             )
```

Comment:
Is there any way we can make this more Pythonic, e.g.:
```
for filter in client_side_filters:
   result = self._apply_client_side_filter(filter)
```

That way we don't care if there's 0 or >=1 elements in the list